src/history/hs_rec.c

```diff
@@ -806,47 +835,30 @@ __wt_hs_delete_key(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btre
 }
 
 /*
- * __hs_delete_reinsert_from_pos --
- *     Delete updates in the history store if the start timestamp of the update is larger or equal
- *     to the specified timestamp and optionally reinsert them with ts-1 timestamp. This function
- *     works by looking ahead of the current cursor position for entries for the same key, removing
- *     them.
+ * __hs_cursor_pos --
+ *     Get the cursor position by looking ahead to find the entries with the same key.
  */
 static int
-__hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btree_id,
-  const WT_ITEM *key, wt_timestamp_t ts, bool reinsert, bool no_ts_tombstone,
-  bool error_on_ts_ordering, uint64_t *counter, WT_TIME_WINDOW *upd_tw)
+__hs_cursor_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btree_id,
+  const WT_ITEM *key, wt_timestamp_t ts, bool error_on_ts_ordering, WT_TIME_WINDOW *upd_tw,
+  bool *non_ts_updates)
 {
-    WT_CURSOR *hs_insert_cursor;
-    WT_CURSOR_BTREE *hs_cbt;
     WT_DECL_RET;
-    WT_ITEM hs_key, hs_value;
-    WT_TIME_WINDOW hs_insert_tw, *twp;
-    wt_timestamp_t hs_durable_start_ts, hs_durable_stop_ts, hs_start_ts;
-    uint64_t cache_hs_order_lose_durable_timestamp, cache_hs_order_reinsert, cache_hs_order_remove;
-    uint64_t hs_counter, hs_upd_type;
+    WT_ITEM hs_key;
+    WT_TIME_WINDOW *twp;
+    wt_timestamp_t hs_start_ts;
+    uint64_t hs_counter;
     uint32_t hs_btree_id;
 #ifdef HAVE_DIAGNOSTIC
     int cmp;
 #endif
-    char ts_string[5][WT_TS_INT_STRING_SIZE];
-
-    hs_insert_cursor = NULL;
-    hs_cbt = __wt_curhs_get_cbt(hs_cursor);
     WT_CLEAR(hs_key);
-    WT_CLEAR(hs_value);
-    cache_hs_order_lose_durable_timestamp = cache_hs_order_reinsert = cache_hs_order_remove = 0;
-
+    WT_UNUSED(btree_id);
+    WT_UNUSED(error_on_ts_ordering);
```

Comment:
ditto, we should be able to remove these WT_UNUSED, if not its worth considering to remove the variables them selves.

Reply:
[2b180e8](https://github.com/wiredtiger/wiredtiger/pull/10686/commits/2b180e83db0c5ee2f52a7de9ce66b5a63924bdf0)